src/libraries/System.Collections/src/System/Collections/Generic/PriorityQueue.cs

```diff
@@ -0,0 +1,695 @@
+        {
+            if (_size == _nodes.Length)
+            {
+                int newCapacity = (int)((long)_nodes.Length * (long)GrowFactor / 100);
+                if (newCapacity < _nodes.Length + MinimumGrow)
+                {
+                    newCapacity = _nodes.Length + MinimumGrow;
+                }
+                SetCapacity(newCapacity);
+            }
+        }
+
+        /// <summary>
+        /// Grows or shrinks the array holding nodes. Capacity must be >= _size.
+        /// </summary>
+        private void SetCapacity(int capacity)
+        {
+            var newArray = new (TElement, TPriority)[capacity];
+
+            if (_size > 0)
+            {
+                Array.Copy(_nodes, 0, newArray, 0, _size);
+            }
+
+            _nodes = newArray;
+        }
+
+        /// <summary>
+        /// Removes the node at the specified index.
+        /// </summary>
+        private void Remove(int indexOfNodeToRemove)
+        {
+            // The idea is to replace the specified node by the very last
+            // node and shorten the array by one.
+
+            int lastNodeIndex = GetLastNodeIndex();
+            var lastNode = _nodes[lastNodeIndex];
+            _size--;
+            _version++;
+
+            // In case we wanted to remove the node that was the last one,
+            // we are done.
+
+            if (indexOfNodeToRemove == lastNodeIndex)
+            {
+                return;
+            }
+
+            // Our last node was erased from the array and needs to be
+            // inserted again. Of course, we will overwrite the node we
+            // wanted to remove. After that operation, we will need
+            // to restore the heap property (in general).
+
+            var nodeToRemove = _nodes[indexOfNodeToRemove];
+
+            int relation = this.Comparer.Compare(lastNode.priority, nodeToRemove.priority);
+            PutAt(lastNode, indexOfNodeToRemove);
+
+            if (relation < 0)
+            {
+                MoveUp(lastNode, indexOfNodeToRemove);
+            }
+            else
+            {
+                MoveDown(lastNode, indexOfNodeToRemove);
+            }
+        }
+
+        /// <summary>
+        /// Puts a node at the specified index.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private void PutAt((TElement element, TPriority priority) node, int index)
+        {
+            _nodes[index] = node;
+        }
+
+        /// <summary>
+        /// Gets the index of the last node in the heap.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private int GetLastNodeIndex() => _size - 1;
+
+        /// <summary>
+        /// Gets the index of an element's parent.
+        /// </summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private int GetParentIndex(int index) => (int)((index - 1) / Arity);
```

Comment:
```suggestion
        private int GetParentIndex(int index) => (int)((uint)(index - 1) / Arity);
```

The operation result of `uint` and` int` is `long`.

before
```asm
    L0000: dec edx
    L0002: movsxd rax, edx
    L0005: mov rdx, rax
    L0008: sar rdx, 0x3f
    L000c: and rdx, 3
    L0010: add rax, rdx
    L0013: sar rax, 2
    L0017: ret
```
after
```asm
    L0000: lea eax, [rdx-1]
    L0003: shr eax, 2
    L0006: ret
```

Reply:
I think it is better to rewrite other expressions related to `Arity` so that they do not become `long`.